src/main/java/appeng/container/implementations/ContainerPatternTerm.java

```diff
@@ -549,44 +548,50 @@ public void setCanBeSubstitute(final boolean beSubstitute) {
         this.beSubstitute = beSubstitute;
     }
 
-    static boolean canDoubleStacks(SlotFake[] slots) {
-        List<SlotFake> enabledSlots = Arrays.stream(slots).filter(SlotFake::isEnabled).collect(Collectors.toList());
-        long emptyStots = enabledSlots.stream().filter(s -> s.getStack() == null).count();
-        long fullSlots = enabledSlots.stream().filter(s -> s.getStack() != null && s.getStack().stackSize * 2 > 127)
-                .count();
-        return fullSlots <= emptyStots && emptyStots < enabledSlots.size();
+    static boolean canDouble(SlotFake[] slots, int mult) {
+        for (Slot s : slots) {
+            if (s.getStack() != null && s.getStack().stackSize * mult < 0) {
```

Comment:
Any reason you use sign check here and Math.multiplyExact later?

Reply:
The reason is I forgot to delete it, mb